src/test/java/uk/gov/hmcts/ccd/v2/external/controller/SearchCaseAssignedUserRolesControllerIT.java

```diff
@@ -0,0 +1,341 @@
+package uk.gov.hmcts.ccd.v2.external.controller;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import uk.gov.hmcts.ccd.MockUtils;
+import uk.gov.hmcts.ccd.v2.V2;
+import uk.gov.hmcts.ccd.v2.external.domain.SearchCaseAssignedUserRolesRequest;
+import uk.gov.hmcts.ccd.v2.external.resource.CaseAssignedUserRolesResource;
+
+import java.util.Arrays;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.ccd.test.RoleAssignmentsHelper.roleAssignmentResponseJson;
+import static uk.gov.hmcts.ccd.test.RoleAssignmentsHelper.userRoleAssignmentJson;
+import static uk.gov.hmcts.ccd.v2.V2.Error.OTHER_USER_CASE_ROLE_ACCESS_NOT_GRANTED;
+
+class SearchCaseAssignedUserRolesControllerIT extends BaseCaseAssignedUserRolesControllerIT {
```

Comment:
The tests in here are the functional equivalents to all those in the existing `GetCaseAssignedUserRolesControllerIT` (which will be deleted eventually), with the only difference being having to call POST with the arguments in the request body instead of query params.